packages/stellar-axelar-std/src/interfaces/upgradable.rs

```diff
@@ -21,17 +21,21 @@ pub trait MigratableInterface: UpgradableInterface + CustomMigratableInterface {
     type Error: Into<soroban_sdk::Error>;
 
     /// Migrates contract state after upgrading the contract code.
-    fn migrate(env: &Env, migration_data: Self::MigrationData) -> Result<(), Self::Error>;
+    fn migrate(
```

Comment:
Can you add comments on how this abstraction works somewhere? I think it'll be hard to understand for someone new to the code or not as familiar with Rust trait features